File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        set<int>people;
        map<int,int>judge;
        if(trust.size()<1 and n==1)
            return n;
        for(auto i : trust){
            people.insert(i[0]);
            judge[i[1]]++;
        }
        if(people.size()!= n-1)
        return -1;
    for (auto it=judge.begin(); it!=judge.end(); ++it) {
        if (people.find(it->first) == people.end() and it->second == people.size() and people.size()!=0)
            return it->first;
    }
    return -1;
    }
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.cs
public class Solution {
    public int FindJudge(int n, int[][] trust){
        
        int[] isTrusted = new int[n+1];
        for (int i = 0; i < trust.Length; i++)
        {
            isTrusted[trust[i][1]]++;
            isTrusted[trust[i][0]]--;
        }
        for (int i = 1; i <=n; i++)
        {
            if (isTrusted[i]==n-1)
                return i;
        }
        return -1;
    }
}